src/daemon/daemon.cpp

```diff
@@ -1340,7 +1343,14 @@ try // clang-format on
     for (const auto& name : instances_to_stop)
     {
         auto it = vm_instances.find(name);
-        it->second->shutdown();
+        if (request->cancel_shutdown())
+        {
+            it->second->start();
```

Comment:
I implemented the timer in the VM since I thought it would be cleaner to always keep a timer object with each VM instead of creating some new unordered_map of <vm_name, QTimer> in the daemon.  But if you feel it's more RAII to keep the timer in the daemon itself, then I'm fine with.  Always different ways to accomplish the same thing:)

Reply:
Right, the main thing was making the shutdown explicit instead of calling start. 

But yeah, one can avoid that all together if the daemon (or an entity owned by the daemon) implements this logic, since the implementation seems to apply to any backend.